src/main/java/software/amazon/nio/spi/s3/S3FileSystem.java

```diff
@@ -42,17 +37,21 @@ public class S3FileSystem extends FileSystem {
      */
     public static final String BASIC_FILE_ATTRIBUTE_VIEW = "basic";
 
-    private final String bucketName;
+    protected S3ClientProvider clientProvider;
```

Comment:
This can probably be private because you have public methods to access and set the client?